src/scores/probability/crps_impl.py

```diff
@@ -1094,3 +1096,101 @@ def _vfunc(x, threshold=threshold):
+    fcst: XarrayLike,
+    obs: XarrayLike,
+    ensemble_member_dim: str,
+    lower_threshold: Union[xr.DataArray, float],
+    upper_threshold: Union[xr.DataArray, float],
+    *,  # Force keywords arguments to be keyword-only
+    method: Literal["ecdf", "fair"] = "ecdf",
+    reduce_dims: Optional[Sequence[str]] = None,
+    preserve_dims: Optional[Sequence[str]] = None,
+    weights: Optional[XarrayLike] = None,
+) -> XarrayLike:
+    """
+    Calculates the threshold weighted continuous ranked probability score (twCRPS)
+    weighted for an interval across the distribution from ensemble input.
+
+    A threshold weight of 1 is assigned for values within the interval and a threshold weight of 0 otherwise.
+    The threshold values that define the bounds of the interval are given by the
+    ``lower_threshold`` and ``upper_threshold`` arguments.
+    For example, if we only want to foucs on the temperatures between -10 and -20 degrees C
+    where aircraft icing is most likely, we can set ``lower_threshold=-20`` and ``upper_threshold=-10``.
+
+
+    For more flexible weighting options and the relevant equations, see the
+    :py:func:`scores.probability.tw_crps_for_ensemble` function.
+
+    Args:
+        fcst: Forecast data. Must have a dimension `ensemble_member_dim`.
+        obs: Observation data.
+        ensemble_member_dim: the dimension that specifies the ensemble member or the sample
+            from the predictive distribution.
+        lower_threshold: the threshold value for where the interval begins. It can either be a float
+            for a single threshold or an xarray object if the threshold varies across
+            dimensions (e.g., climatological values).
+        upper_threshold: the threshold value for where the interval ends. It can either be a float
+            for a single threshold or an xarray object if the threshold varies across
+            dimensions (e.g., climatological values).
+        method: Either "ecdf" or "fair". See :py:func:`scores.probability.tw_crps_for_ensemble`
+            for more details.
+        reduce_dims: Dimensions to reduce. Can be "all" to reduce all dimensions.
+        preserve_dims: Dimensions to preserve. Can be "all" to preserve all dimensions.
+        weights: Weights for calculating a weighted mean of individual scores. Note that
+            these weights are different to threshold weighting which is done by decision
+            threshold.
+
+    Returns:
+        xarray object of twCRPS values that has been weighted based on an interval.
+
+    Raises:
+        ValueError: when ``lower_threshold`` is not less than ``upper_threshold``.
+        ValueError: when ``method`` is not one of "ecdf" or "fair".
+
+    References:
+        Allen, S., Ginsbourger, D., & Ziegel, J. (2023). Evaluating forecasts for high-impact
+        events using transformed kernel scores. SIAM/ASA Journal on Uncertainty
+        Quantification, 11(3), 906-940. https://doi.org/10.1137/22M1532184
```

Comment:
Are there any good open access links for further reading that could be added? Okay if not, just checking.

Reply:
If you google the name of the paper it will come up on arXiv.  @Steph-Chong  raised this question on a previous pull request.

Most of the papers that we cite in scores aren't open access. I think it would be good to discuss if we add additional links to freely available PDFs everywhere in scores, or leave the onus on the user to google the paper name to find a PDF.

Reply:
Oh - I see that you to recommend that we do have open access links in https://github.com/nci/scores/pull/690/files#diff-bce1a2622557d54c6b5f5e1f6ae398c1cec86fdd0ae8b3a32208bc6be77d1231R46

If we want to do this, someone needs to go through and add open access links for most of the metrics!

Reply:
I've added an additional sentence to the end of the reference. It deviates from the referencing style, so let me know what you think @Steph-Chong and @tennlee

Reply:
I don't think linking into republishing sites which host the original PDF but not via the original publisher is a good idea (even if it's a legitimate thing) - I think we can leave that kind of searching to users. I also think it's good practice to provide the proper citation to subscription-access materials. However, sometimes there are quite good open access articles which are a review, or a discussion of scores, or useful further reading. Even just a wikipedia link if the article is good. So it's more about providing a suggested alternative to people -- but -- only if the suggested alternative is actually worthwhile, not to be to prescriptive about it. I'll try to improve the wording in the draft template. In terms of going through past work and adding links, I think it could be nice, but it's not a top priority right now.

In terms of this PR, I'm not sure that interval twCRPS is likely to have much by way of open access material, but I thought I'd just mention it. I think it's fine to leave it for now unless you happen to be aware of good open access further reading.

Reply:
Okay, I've found an open source reference we can use. https://www.jstatsoft.org/article/view/v110i08 (it only came out around a week ago)

Reply:
https://www.jstatsoft.org/article/view/v110i08

Reply:
I agree with @tennlee's take on references that he outlined above. Even if we haven't always done so in the past, going forward I think it would be good to add open access references when there is a good one available. (And I agree, open access references should only be included when they are worthwhile, and not otherwise).

In terms of the existing references in `scores`, I have been gradually changing non-DOI URLs to DOIs (when available). I have been doing this here and there as I get a chance, so it has been occurring slowly.

In some instances, the initially provided URL was to a pay-walled third-party republishing site (e.g. JSTOR), when the article was freely available via the publisher issued DOI - I think I have now fixed a lot of these, but there may still be some left. 

I realise in plenty of instances, changing the non-DOI URL to a DOI won't make the article freely available to readers, but where it does, that's a handy benefit.

For existing metrics without an open access reference, I agree actively going and adding open access references isn't currently a top priority. But if/when it can be done opportunistically, then we should (just like @nicholasloveday  has done here, with adding the (newly published) open access article to existing twCRPS entries).